lib/Bio/EnsEMBL/DataCheck/Checks/MemberProductionCounts.pm

```diff
@@ -0,0 +1,204 @@
+  
+  my $sql_sums = qq/
+    SELECT SUM(families) AS sum_families, SUM(gene_trees) AS sum_gene_trees, SUM(gene_gain_loss_trees) AS sum_gene_gain_loss_trees, SUM(orthologues) AS sum_orthologues, SUM(paralogues) AS sum_paralogues, SUM(homoeologues) AS sum_homoeologues 
+      FROM gene_member_hom_stats 
+    WHERE collection = ?
+  /;
+
+  my $sqlCollections = qq/
+     SELECT DISTINCT clusterset_id 
+       FROM gene_tree_root 
+     WHERE $constraint
+   /;
+
+  my $collections = $helper->execute_simple( -SQL => $sqlCollections );
+
+  my $sqlFamilies = q/
+    SELECT COUNT(*) 
+      FROM family
+  /; # To be tested only with default
+  my $family_count = $helper->execute_single_result( -SQL => $sqlFamilies );
+
+  my $sqlPolyploids = q/
+    SELECT COUNT(*) 
+      FROM genome_db 
+    WHERE genome_component IS NOT NULL
+  /; # Assumes that the polyploid genomes are found in all the collections
+  my $polyploid_count = $helper->execute_single_result( -SQL => $sqlPolyploids );
+  
+  my $sqlGeneTrees = qq/
+    SELECT COUNT(*) 
+      FROM gene_tree_root 
+    WHERE clusterset_id = ?
+  /;
+  
+  my $sqlCAFETrees = qq/
+    SELECT COUNT(*) 
+      FROM gene_tree_root 
+        JOIN CAFE_gene_family 
+          ON gene_tree_root.root_id = gene_tree_root_id 
+    WHERE clusterset_id = ?
+  /;
+  
+  my @columns = qw(families gene_trees gene_gain_loss_trees orthologues paralogues homoeologues);
+
+  foreach my $collection ( @$collections ) {
+
+    my $sums = $helper->execute( -SQL => $sql_sums, -USE_HASHREFS => 1, PARAMS => [$collection] );
+
+    my $genetree_count = $helper->execute_single_result( -SQL => $sqlGeneTrees,  PARAMS => [$collection] );
+    my $cafetrees_count = $helper->execute_single_result( -SQL => $sqlCAFETrees, PARAMS => [$collection] );
+    
+    my @counts = ($family_count, $genetree_count, $cafetrees_count, $genetree_count, $genetree_count, $genetree_count, $polyploid_count);
+
+    if ( $dba->get_division() =~ /vertebrates/ && $collection =~ /default/ ) {
+      my $desc_5 = "The sum of entries for families in gene_member_hom_stats > 0 for the $collection collection";
+      cmp_ok( $sums->[0]->{sum_families}, ">", 0, $desc_5 );
+    }
+    if ( $dba->get_division() =~ /vertebrates/ || $collection =~ /default/ ) {
+      my $desc_6 = "The sum of entries for gene_trees in gene_member_hom_stats > 0 for the $collection collection";
+      cmp_ok( $sums->[0]->{sum_gene_trees}, ">", 0, $desc_6 );
+    }
+
+    foreach ( my $i = 0; $i < @columns; $i++ ) {
+      my $desc_7 = "Found expected entries in gene_member_hom_stats with $columns[$i] > 0 for the $collection collection";
+      my $desc_8 = "There were no unexpected entries in gene_member_hom_stats with $columns[$i] > 0 for the $collection collection";
+      next if ( $collection !~ /default/ && $columns[$i] eq "families" );
+      next if ( $dba->get_division() !~ /plants/ && $columns[$i] eq "homoeologues" );
```

Comment:
I think this test is superfluous. The "homoeologues" column seems to be controlled by `$polyploid_count`, which is the more generic approach

Reply:
Generally speaking, I would prefer to minimize the number of times we check the division name in the DCs. They could perhaps rely on the mlss table instead, with a separate DC to check that all the MLSS types are found given the division name. This would ideally be the only DC that is division-specific. Will put that into a ticket to review the DCs once all the PRs are merged